### PR TITLE
more frontend refactoring

### PR DESCRIPTION
**Pull request by livnekes**

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,11 @@
-# Use the official image as a parent image.
-FROM node:current-slim
+FROM nginx
```
Comment (frontegg-david) [inline frontend/Dockerfile:1]: `FROM nginx:1.19-alpine`

File: frontend/Dockerfile [hunk 2]
```diff
@@ -1,19 +1,11 @@
...
 
-# Set the working directory.
 WORKDIR /app
 
-# add `/app/node_modules/.bin` to $PATH
-ENV PATH /app/node_modules/.bin:$PATH
+COPY ./nginx.conf /etc/nginx/conf.d/default.conf
```
Comment (frontegg-david) [inline frontend/Dockerfile:5]: `COPY ./nginx.conf /etc/nginx/nginx.conf`

File: frontend/Dockerfile (hunk 1 above)
Comment (frontegg-david) [inline frontend/Dockerfile:1]: `FROM nginx:1.19-alpine`

File: frontend/Dockerfile [hunk 3]
```diff
@@ -1,19 +1,11 @@
-# Use the official image as a parent image.
-FROM node:current-slim
+FROM nginx
 
-# Set the working directory.
 WORKDIR /app
```
Comment (frontegg-david) [inline frontend/Dockerfile:3]: you can remove

File: frontend/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,11 @@
+server {
+    listen 80;
+    server_name _;
+
+    root /var/www/;
```
Comment (frontegg-david) [inline frontend/nginx.conf:5]: check this root

File: frontend/nginx.conf [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    index index.html;
+
+    location / {
+        try_files $uri /index.html;
```
Comment (frontegg-david) [inline frontend/nginx.conf:9]: why using /index.html, replace with ./index.html or index.html

File: frontend/src/Components/DatePicker/DatePicker.tsx [hunk 6]
```diff
@@ -5,7 +5,7 @@ import './DatePicker.scss';
 type DatePickerProps = {
   type?: string;
   header: string;
```
Comment (frontegg-david) [inline frontend/src/Components/DatePicker/DatePicker.tsx:7]: inline: boolean

File: frontend/src/Components/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -1,55 +1,50 @@
...
-type Props = {
-  first?: JSX.Element | string;
-  last?: JSX.Element | string;
-  prev?: JSX.Element | string;
-  next?: JSX.Element | string;
-  numberComponent?: React.ComponentType<any> | React.ElementType<any>;
-  numberRender?: (props: React.HTMLAttributes<HTMLElement>) => JSX.Element;
+interface PaginationProps extends React.HTMLAttributes<HTMLElement>, UsePaginationProps {
+  first?: React.ReactElement | string;
+  last?: React.ReactElement | string;
```
Comment (frontegg-david) [inline frontend/src/Components/Pagination/Pagination.tsx:8]: change `React.ReactElement | string` to `React.ReactNode`

File: frontend/src/Pages/MainLayout/DatePickerPage/DatePickerPage.tsx [hunk 8]
```diff
@@ -4,19 +4,18 @@ import DatePicker from '../../../Components/DatePicker';
 const DatePickerPage: React.FC = () => {
   const inlineDatepicker = [
     {
-      type: 'InlineDatepicker',
+      type: 'inline',
```
Comment (frontegg-david) [inline frontend/src/Pages/MainLayout/DatePickerPage/DatePickerPage.tsx:7]: change to inline without type

File: frontend/src/Pages/MainLayout/DatePickerPage/DatePickerPage.tsx [hunk 9]
```diff
@@ -4,19 +4,18 @@ import DatePicker from '../../../Components/DatePicker';
...
       header: 'Inline date picker',
       description: (
         <span>
-          Use the <code> &lt;InlineDatepicker /&gt;</code> component to display a datepicker
+          Use the <code> &lt;Datepicker type='inline'/&gt;</code> component to display a datepicker
```
Comment (frontegg-david) [inline frontend/src/Pages/MainLayout/DatePickerPage/DatePickerPage.tsx:11]: change to inline without type

**Review by frontegg-david — CHANGES_REQUESTED** [review]

File: frontend/nginx.conf (hunk 5 above)
Comment (livnekes) [reply frontend/nginx.conf:9]: doesn't work without the slash. don't know why ...

**Review by frontegg-david — APPROVED** [review]

**Pull request merged**